pkg/volume/metadata/metadata.go

```diff
@@ -0,0 +1,255 @@
+	Name:         "not-used",
+	VolumeSource: api.VolumeSource{EmptyDir: &api.EmptyDirVolumeSource{Medium: api.StorageTypeMemory}},
+}
+
+// createRandomSymlink try to create a temporary symbolic link
+func createRandomSymlink(newTarget string) (string, error) {
+	var runes = []rune("abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ")
+	b := make([]rune, 5)
+	for j := range b {
+		b[j] = runes[rand.Intn(len(runes))]
+	}
+	randomSymLink := string(b)
+	if e := os.Symlink(newTarget, randomSymLink); e == nil {
+		return randomSymLink, e
+	}
+	return "", fmt.Errorf("unable to create random symlink")
+}
+
+// readFile reads full path file name and return it as a string
+func readFile(fileName string) string {
+	if data, err := ioutil.ReadFile(fileName); err == nil {
+		return string(data)
+	}
+	return ""
+}
+
+// generateFile writes input parameter 'values' in a new file.
+// It creates a '.tmp.' prefixed temporary file and renames it afterwards
+func generateFile(values string) (string, error) {
+	const tmpPrefix = ".tmp."
+	prefix := tmpPrefix + time.Now().Format("01_01_010101")
+	tmpFile, err := ioutil.TempFile(".", prefix)
+	if err != nil {
+		return "", fmt.Errorf("unable to create tmp file '%s', with prefix '%s'", tmpFile.Name(), prefix)
+	}
+	if e := ioutil.WriteFile(tmpFile.Name(), []byte(values), 0644); e != nil {
+		return "", e
+	}
+	_, f := path.Split(tmpFile.Name())
+	newFileName := strings.TrimPrefix(f, tmpPrefix)
+	if e := os.Rename(tmpFile.Name(), newFileName); e != nil {
+		return "", e
+	}
+	return newFileName, nil
+}
+
+// isFileToGenerate compares actual file with the new values. If
+// different (or the file does not exist) return true,
+func isFileToGenerate(fileName, values string) bool {
+	return readFile(fileName) != values
+}
+
+// WriteFile writes string in fileName file if string must be updated.
+// It writes 'atomically' writing in a tmp file and re-placing the previous copy
+// via symbol linking recreation. It ends up removing the previous file
+func WriteFile(fileName, dirPath, values string) error {
+	if values == "" || fileName == "" {
+		return nil
+	}
+	// normalize path
+	dir, file := path.Split(path.Join(dirPath, fileName))
+	wd, e := os.Getwd()
+	if e != nil {
+		return e
+	}
+	defer os.Chdir(wd)
+	// to simplify  enter in dir where files and link are generated.
+	if dir != "" {
+		if _, err := os.Stat(dir); err != nil {
+			return err
+		}
+		if e := os.Chdir(dir); e != nil {
+			return e
+		}
+	}
+	// check wether file has be to generated
+	if !isFileToGenerate(file, values) {
+		return nil
+	}
+	newFileName, err := generateFile(values)
+	if err != nil {
+		return err
+	}
+	actualTarget, e := os.Readlink(file)
+	if e != nil { // no link so creates one
+		os.Symlink(newFileName, file)
+		return nil
+	}
+	newLink, e := createRandomSymlink(newFileName)
+	if e != nil {
+		return e
+	}
+	if e := os.Rename(newLink, file); e != nil {
+		return e
+	}
+	return os.Remove(actualTarget)
+}
+
```

Comment:
I think this func should have godoc explaining the scheme for the contents of the directory, ie, which files and symlinks exist, how are the links updated, etc.